src/all/pornpics/src/eu/kanade/tachiyomi/extension/all/pornpics/PornPicsExt.kt

```diff
@@ -0,0 +1,49 @@
+package eu.kanade.tachiyomi.extension.all.pornpics
+
+import okhttp3.HttpUrl
+
+internal const val QUERY_PAGE_SIZE = 19
+
+internal fun HttpUrl.Builder.addQueryParameter(encodedName: String, encodedValue: Int) =
+    addQueryParameter(encodedName, encodedValue.toString())
+
+internal fun HttpUrl.Builder.addQueryPageParameter(page: Int) =
+// Add +1 to requested image count per page,
+    // Compare actual received count with pageSize to determine next page.
+    this.addQueryParameter("limit", QUERY_PAGE_SIZE + 1)
+        .addQueryParameter("offset", (page - 1) * QUERY_PAGE_SIZE)
+
+internal fun HttpUrl.Builder.addUrlPart(
+    urlPart: String?,
+    addPath: Boolean = true,
+    addQuery: Boolean = true,
+): HttpUrl.Builder {
```

Comment:
Can you not access the original `HttpUrl` object from a builder? Should prefer their parsing for query parameters.